qiita_ware/ebi.py

```diff
@@ -77,56 +80,74 @@ def iter_file_via_list_of_dicts(input_file):
 
 
 class EBISubmission(object):
-    """Define an EBI submission and facilitate generation of required XML files
+    """Define an EBI submission, generate submission files and submit
+
+    Submit a preprocessed data to EBI
+
+    The steps for EBI submission are:
+    1. Validate that the submission status is different than
+    self.valid_ebi_actions and that there is a valid investigation_type
+    2. Generate per sample demultiplexed files
+    3. Generate XML files for submission
+    4. Submit sequences files
+    5. Submit XML files. The answer has the EBI submission numbers.
+
+    Parameters
+    ----------
+    preprocessed_data_id : int
+        The preprocesssed data id
+    action : str
+        The action to perfom, it has to be one of the self.valid_ebi_actions
 
     Parameters
     ----------
     preprocessed_data_id : str
-    study_title : str
-    study_abstract : str
-    investigation_type : str
-        Any of the options provided by ebi, see:
-        https://www.ebi.ac.uk/ega/sites/ebi.ac.uk.ega/files/documents/Study.xml
-        If `'Other'` is passed, then you must provide a value in the
-        new_investigation_type argument.
-    empty_value : str, optional
-        Defaults to "no_data". This is the value that will be used when data
-        for a particular metadata field is missing
-    new_investigation_type: str, optional
-        'metagenome', and 'mimarks-survey' are specially recognized and used to
-        set other attributes in the submission, but any string is valid. This
-        value is required only if `'Other'` is passed in `investigation_type`.
     """
-    def __init__(self, preprocessed_data_id, study_title, study_abstract,
-                 investigation_type, empty_value='no_data',
-                 new_investigation_type=None, pmids=None, **kwargs):
+    def __init__(self, preprocessed_data_id, action, **kwargs):
+        self.valid_ebi_actions = ('ADD', 'VALIDATE', 'MODIFY')
+        self.valid_ebi_submission_states = ('submitting', 'success')
+
+        # Step 1: variable setup and validations
+        if action not in self.valid_ebi_actions:
+            raise ValueError("Not a valid action (%s): %s" % (
+                ', '.join(self.valid_ebi_actions), action))
+
+        ena_ontology = Ontology(convert_to_id('ENA', 'ontology'))
+        ppd = PreprocessedData(preprocessed_data_id)
+        s = Study(ppd.study)
+        pt = PrepTemplate(ppd.prep_template)
+
+        status = ppd.submitted_to_insdc_status()
+        if status in self.valid_ebi_submission_states:
```

Comment:
The current code doesn't allow any modification/validation, really. The only thing that changes is what's added in the text. The "right" way to do it will be to only update the metadata on modify (you can not modify seq files automatically).